### PR TITLE
Stale bot updated

### PR DESCRIPTION
**Pull request by LysandreJik**
This is an updated version of the stale bot.

**It is easier to review the file than the diff, you can find the file [here](https://github.com/huggingface/transformers/blob/d1e516ea0fe9e641a75a89e5a7522392f7dbd59d/scripts/stale.py).**

It sends a warning message after 23 days of inactivity, and closes the issue/PR if no activity is detected in the following 7 days.

It ignores the following labels (case insensitive):
- `Good First Issue`
- `Good Second Issue`
- `Feature Request`
- `New Model`
- `WIP`

If there are assignees on the issue/PR, then it puts the following comment: `f"This issue has been stale for a while, ping @{assignee.login}"`

I propose to leave the PR like it is, and I'll push an empty commit daily to check the result of the stale bot test (I'll remove other tests to ensure that we don't spend unnecessary CI credits). Once we verify that it works as expected for a few days, I propose to merge it.

You may check the results of the first run here: https://github.com/huggingface/transformers/runs/2045189559?check_suite_focus=true (Second commit was rate limited)

### CODE REVIEW

File: scripts/stale.py [hunk 1]
```diff
@@ -35,32 +36,40 @@ def main():
...
+            and not any(label.name.lower() in LABELS_TO_EXEMPT for label in issue.get_labels())
         ):
             for assignee in issue.assignees:
-                print(f"Issue {issue.number}. Pinging {assignee.name} with message")
-                print(f"Hey @{assignee.login}, could you take a second look at this issue?")
-
-                # issue.create_comment(
-                #    f"Hey @{assignee.login}, could you take a second look at this issue?"
-                # )
+                print(f"This issue ({issue.number}) has been stale for a while, ping @{assignee.login}")
```
Comment (stas00) [inline scripts/stale.py:71]: Thank you, @LysandreJik 
I only have a comment about:

> If there are assignees on the issue/PR, then it puts the following comment: f"This issue has been stale for a while, ping @{assignee.login}"

This creates unnecessary noise. e.g. one Issue that had 2 assignees received this comment twice for each assignee and we were notified twice as well on our own and each other's warning.

What I'm proposing is:
1. there should be only one comment issued regardless of how many assignees are 
2. To solve (1) easier I am not sure there is a need to mention the assignee at all - i.e. no need to make a special case for Issues with assignees - whatever the general message is sent is good enough. The assignee gets it anyway - I don't think an assignee can unsubscribe (I could be wrong), so the tagging doesn't add anything since the person is already subscribed.

Bottom line - if the subscribed person with the right perms wants to do something about the Issue/PR they can. If they don't tagging won't do anything useful OMHO.

**Review by stas00 — APPROVED** [review]
Thank you, @LysandreJik

Just one issue - commented below

File: scripts/stale.py (hunk 1 above)
Comment (LysandreJik) [reply scripts/stale.py]: Let's go with 2a! Thanks for your comment

**Comment by LysandreJik** [comment]
@stas00, @sgugger, please review this PR. Here is a visualization of what would be done by the bot, were it to be merged today: https://github.com/huggingface/transformers/pull/10562/checks?check_run_id=2113061781

I have verified that all 11 issues that would be closed have received a warning 10 days ago. Thank you.

File: scripts/stale.py [hunk 2]
```diff
@@ -35,32 +36,29 @@ def main():
...
-            and (dt.utcnow() - issue.updated_at).days > 21
+            last_comment is not None and last_comment.user.login == "github-actions[bot]"
+            and (dt.utcnow() - issue.updated_at).days > 7
             and (dt.utcnow() - issue.created_at).days >= 30
             and not any(label.name.lower() in LABELS_TO_EXEMPT for label in issue.get_labels())
         ):
-            print("Closing", issue)
-            # issue.create_comment(
-            #     "This issue has been automatically marked as stale and been closed because it has not had "
-            #     "recent activity. Thank you for your contributions.\n\nIf you think this still needs to be addressed"
```
Comment (sgugger) [inline scripts/stale.py:47]: This would need to be adapted with the same message as in the print statement when uncommented.

File: scripts/stale.py [hunk 3]
```diff
@@ -35,32 +36,29 @@ def main():
...
         elif (
-            len(issue.assignees) > 0
-            and (dt.utcnow() - issue.updated_at).days > 21
+            (dt.utcnow() - issue.updated_at).days > 23
             and (dt.utcnow() - issue.created_at).days >= 30
+            and not any(label.name.lower() in LABELS_TO_EXEMPT for label in issue.get_labels())
         ):
-            for assignee in issue.assignees:
-                print(f"Issue {issue.number}. Pinging {assignee.name} with message")
-                print(f"Hey @{assignee.login}, could you take a second look at this issue?")
```
Comment (sgugger) [inline scripts/stale.py:59]: Can we add a message to `assignee.login` to tell him how to stop this (unassign themselves for instance)? The bot is going to give the same warning 23 days later if there is nothing to do on it.

**Review by sgugger — APPROVED** [review]
LGTM, thanks for working on this!

File: scripts/stale.py (hunk 2 above)
Comment (LysandreJik) [reply scripts/stale.py:47]: Actually, I had in mind to leave it like this. I think the message as it is is way better than the message in the print statement, which was here only for verification purposes.

File: scripts/stale.py (hunk 3 above)
Comment (LysandreJik) [reply scripts/stale.py:59]: There isn't any specific behavior when there is an assignee anymore, as the assignee will still get a ping on the issue all the same. The assignee isn't going to receive the same warning 23 days later, the issue is going to be closed 7 days later.

File: scripts/stale.py [hunk 4]
```diff
@@ -35,32 +36,29 @@ def main():
...
             and not any(label.name.lower() in LABELS_TO_EXEMPT for label in issue.get_labels())
         ):
-            print("Closing", issue)
-            # issue.create_comment(
-            #     "This issue has been automatically marked as stale and been closed because it has not had "
-            #     "recent activity. Thank you for your contributions.\n\nIf you think this still needs to be addressed"
-            #     " please comment on this thread."
-            # )
-            # issue.add_to_labels("wontfix")
+            print(f"Would close issue {issue.number} since it has been 7 days of inactivity since bot mention.")
```
Comment (stas00) [inline scripts/stale.py:47]:
```suggestion
            print(f"Closing issue {issue.number} since it has been 7 days of inactivity since bot mention.")
```

"would close" was probably more suitable for debug ;)

File: scripts/stale.py [hunk 5]
```diff
@@ -35,32 +36,29 @@ def main():
...
+            and not any(label.name.lower() in LABELS_TO_EXEMPT for label in issue.get_labels())
         ):
-            for assignee in issue.assignees:
-                print(f"Issue {issue.number}. Pinging {assignee.name} with message")
-                print(f"Hey @{assignee.login}, could you take a second look at this issue?")
-
-                # issue.create_comment(
-                #    f"Hey @{assignee.login}, could you take a second look at this issue?"
-                # )
+            print(f"Would add stale comment to {issue.number}")
```
Comment (stas00) [inline scripts/stale.py:54]:
```suggestion
            print(f"Adding stale comment to {issue.number}")
```

same as with "would close"?

**Review by stas00 — APPROVED** [review]
Looks great, @LysandreJik !

File: scripts/stale.py (hunk 4 above)
Comment (LysandreJik) [reply scripts/stale.py:47]: The print statements were actually only here for debugging purposes, I was thinking I'd remove them just before merging, at the same time uncommenting the actual action.

If you'd rather keep the prints for debugging purposes, that's not an issue, I can leave them.

Comment (stas00) [reply scripts/stale.py:47]: oh, right, there are debug ;) 

then it doesn't matter. I'd just comment them out rather than delete for an easier debug in the future should something go wrong ;)
